packages/html-reporter/src/reportView.tsx

```diff
@@ -85,19 +85,30 @@ export const ReportView: React.FC<{
         />
       </Route>
       <Route predicate={testCaseRoutePredicate}>
-        {!!report && <TestCaseViewLoader report={report} tests={filteredTests.tests} testIdToFileIdMap={testIdToFileIdMap} />}
+        {!!report ?
+          <TestCaseViewLoader report={report} tests={filteredTests.tests} testIdToFileIdMap={testIdToFileIdMap} /> :
+          <div className='error'>Report data could not be found</div>}
       </Route>
     </main>
   </div>;
 };
 
+type TestState = {
+  type: 'loading'
+} | {
+  type: 'test',
+  test: TestCase,
+} | {
+  type: 'error',
+};
+
 const TestCaseViewLoader: React.FC<{
   report: LoadedReport,
   tests: TestCaseSummary[],
   testIdToFileIdMap: Map<string, string>,
 }> = ({ report, testIdToFileIdMap, tests }) => {
   const searchParams = React.useContext(SearchParamsContext);
-  const [test, setTest] = React.useState<TestCase | undefined>();
+  const [test, setTest] = React.useState<TestState>({ type: 'loading' });
```

Comment:
It is immediately harder to read. Let's keep things more direct. We used to have a test: TestCase, understand that. We can handle undefined test case, we can detect if the test is not the the map and handle that. Where is the state coming from?